### ISSUE TITLE
Xcode 15 Beta 8 + Xcode Cloud Error

### ISSUE THREAD

**Issue by dcilia** [realm/SwiftLint#5205]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Hello.  We are utilizing the SwiftLint plugin as a build phase in our project containing:

- iOS App Target
- iOS App Clip Target  

When building with Xcode Cloud using Xcode Beta (8 in this case) a build error is encountered related to SwiftLint.

```
2023-09-05T18:13:05.391803418Z	2023-09-05 11:12:56.500 xcodebuild[10687:38398]  DVTFilePathFSEvents: Failed to start fs event stream.
2023-09-05T18:13:05.391812273Z	Error: You don’t have permission to save the file “f50a50e51ef5bdd780b98209347c4360c491f7299bcc424e3316162e78de3c24.plist” in the folder “478B1767-7422-3827-9D90-A09D769C708F”.
2023-09-05T18:13:05.391819938Z	
2023-09-05T18:13:05.391824842Z	/usr/bin/sandbox-exec -p "(version 1)
2023-09-05T18:13:05.391830105Z	(deny default)
2023-09-05T18:13:05.391849244Z	(import \"system.sb\")
2023-09-05T18:13:05.391854386Z	(allow file-read*)
2023-09-05T18:13:05.391859099Z	(allow process*)
2023-09-05T18:13:05.391863573Z	(allow file-write*
2023-09-05T18:13:05.391886045Z	    (subpath \"/private/tmp\")
2023-09-05T18:13:05.391896318Z	    (subpath \"/private/var/folders/x7/fs459gdd67132_hw5t3cz_s00000gn/T\")
2023-09-05T18:13:05.391913502Z	)
2023-09-05T18:13:05.391918327Z	(deny file-write*
2023-09-05T18:13:05.391923035Z	    (subpath \"/Volumes/workspace/repository\")
2023-09-05T18:13:05.391928353Z	)
2023-09-05T18:13:05.391933923Z	(allow file-write*
2023-09-05T18:13:05.391938967Z	    (subpath \"/Volumes/workspace/DerivedData/SourcePackages/plugins/**REDACTED**.output/**REDACTED**/SwiftLintPlugin\")
```

```
2023-09-05T18:13:05.453548242Z	2023-09-05 11:12:56.602 xcodebuild[10689:38404]  DVTFilePathFSEvents: Failed to start fs event stream.
2023-09-05T18:13:05.453557706Z	Error: You don’t have permission to save the file “f50a50e51ef5bdd780b98209347c4360c491f7299bcc424e3316162e78de3c24.plist” in the folder “478B1767-7422-3827-9D90-A09D769C708F”.
```

Additionally, our `ci_post_clone` script contains `defaults write com.apple.dt.Xcode IDESkipPackagePluginFingerprintValidatation -bool YES`

Xcode Cloud Environment is set to:

- Latest or Beta Release Xcode
- Latest or Beta Release macOS

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
let foo = try! bar()
```

**Comment by Timac** [comment]
We are seeing the same issue with Xcode 15 RC (15A240d) and macOS Sonoma 14 RC (23A339) on Xcode Cloud.

**Comment by jdelaune** [comment]
Facing the same issue.

Doesn't matter if we use different macOS versions so must be something in Xcode 15.

**Comment by mrgrauel** [comment]
I'm also facing the same issue. I have Swiftlint installed as plugin.
It works locally.

**Comment by KoCMoHaBTa** [comment]
Same issue here

**Comment by phranck** [comment]
Same issue here

**Comment by honus** [comment]
Ran into this as well. The only way I could get it to work was to hack swiftlint so isLikelyXcodeCloudEnvironment always returned true (and jump through the hoops to host my own binary of it). I couldn't figure out how to get logs out of it when running on Xcode Cloud so I'm not sure why that function isn't returning true when it should.  As far as I can tell all the right things are in the environment.

**Comment by simba909** [comment]
@honus This made me think about and look at the docs for Xcode Cloud environment variables, at which point I noticed this entry that I don't think I've seen before:

> **CI_XCODE_CLOUD**
> A Boolean value that evaluates to TRUE when the custom build script runs in Xcode Cloud. Use it to only run the script content when Xcode Cloud builds your project or workspace.

Still haven't looked at why `isLikelyXcodeCloudEnvironment` isn't working (which it still should, I think) but maybe it could be simplified now with the new(?) variable? 🤔

**Comment by honus** [comment]
Could be. I haven't figured out how to get logs out of swiftlint running in Xcode Cloud so it is hard to diagnose.

**Comment by GeekOnIce** [comment]
Having this problem as well, and I'm guessing the reason that [isLikelyXcodeCloudEnvironment](https://github.com/realm/SwiftLint/blob/0934c18cba3de323033920c1d0c2b38399e2745a/Source/swiftlint/Extensions/ProcessInfo%2BXcodeCloud.swift#L4) is no longer working has something to do with the `CI_WORKSPACE` environment variable it references being deprecated. In my custom scripts that I haven't had a chance to update yet, I'm seeing the following warning: `Found usage of deprecated environment variable CI_WORKSPACE. Use CI_PRIMARY_REPOSITORY_PATH instead`

It doesn't quite make sense why that `isLikelyXcodeCloudEnvironment` would fail due to this deprecation though - from everything I can see that environment variable is still set, and the `isLikelyXcodeCloudEnvironment` simply checks for it's existence - but it does seem a bit _too_ coincidental to not be related.

**Comment by piotrtobolski** [comment]
Same issue here, is there any easy workaround for this?

**Comment by merichle** [comment]
We are seeing the same issue with **Xcode 15.0.1 RC (15A507)** and **macOS Sonoma 14.1 RC (23B73)**  and **SwiftLint 0.53.0** on Xcode Cloud.

```
xcodebuild[8854:31114]  DVTFilePathFSEvents: Failed to start fs event stream.
Error: You don’t have permission to save the file “0c1061be49c8d9a08abc57a1d1d91818db5ea359427cb0747c74b61d4fb20b86.plist” in the folder “EE88B62F-83D8-3E14-99E8-23B5D4B2F412”.
```

**Comment by piotrtobolski** [comment]
Any news about this issue?

**Comment by SimplyDanny** [comment]
Fixed by #5287.

**Comment by jaredsinclair** [comment]
This issue does not appear to have been fully resolved by #5287. I'm encountering the following error today:

```
v - Prepare Packages
vv - Running SwiftLint for MyApp 16.0 seconds

/usr/bin/sandbox-exec -p "(version 1)
(deny default)
(import \"system.sb\")
(allow file-read*)
(allow process*)
(allow mach-lookup (global-name \"com.apple.lsd.mapdb\"))
(allow file-write*
    (subpath \"/private/tmp\")
    (subpath \"/private/var/folders/lz/kwk55s8n75j_r0ckr6m6qj9h0000gn/T\")
)
(deny file-write*
    (subpath \"/Volumes/workspace/repository/MyApp\")
)
(allow file-write*
    (subpath \"/Volumes/workspace/DerivedData/SourcePackages/plugins/MyApp.output/MyApp/SwiftLint\")
)
" /Volumes/workspace/DerivedData/SourcePackages/artifacts/MyPrivatePackage/SwiftLintBinary/SwiftLintBinary.artifactbundle/swiftlint-0.54.0-macos/bin/swiftlint lint --strict --quiet --config /Volumes/workspace/repository/MyApp/.swiftlint.yml --cache-path /Volumes/workspace/DerivedData/SourcePackages/plugins/MyApp.output/MyApp/SwiftLint

2024-01-31 09:42:24.142 xcodebuild[6350:24748]  DVTFilePathFSEvents: Failed to start fs event stream.
2024-01-31 09:42:24.447 xcodebuild[6353:24760]  DVTFilePathFSEvents: Failed to start fs event stream.
2024-01-31 09:42:25.584 xcodebuild[6358:24778]  DVTFilePathFSEvents: Failed to start fs event stream.
Error: You don’t have permission to save the file “26c6d282529428ce0cf1ecf2295388c5f0fb2580bd42d99096ebb8d3b99f4da4.plist” in the folder “4D4FAECE-7849-3B2C-9674-8DDA23306B7E”.
```

### Setup:

- SwiftLint 0.54 binary release
- SwiftLint is included as a binary dependency of an internal Swift package `MyPrivatePackage`
- Project target relies on `MyPrivatePackage`'s copy of SwiftLint as a build plugin step under a "Run Build Tool Plugins" Xcode build phase.
- Xcode Cloud (set to "Latest Release" tooling)

### Xcode Cloud Environment

The Xcode Cloud environment variables during Xcode Cloud's "Set environment variables" build step are:

```
CI=TRUE
CI_BRANCH=feature/jared/my-branch-name
CI_BUILD_ID=4a399832-e8ee-45cb-970a-350c55b478ce
CI_BUILD_NUMBER=21
CI_BUILD_URL=[REDACTED]
CI_BUNDLE_ID=[REDACTED]
CI_COMMIT=[REDACTED]
CI_DERIVED_DATA_PATH=/Volumes/workspace/DerivedData
CI_GIT_REF=refs/heads/feature/jared/my-branch-name
CI_PRIMARY_REPOSITORY_PATH=/Volumes/workspace/repository
CI_PRODUCT=MyApp
CI_PRODUCT_ID=[REDACTED]
CI_PRODUCT_PLATFORM=iOS
CI_PROJECT_FILE_PATH=/Volumes/workspace/repository/MyApp/MyApp.xcworkspace
CI_PULL_REQUEST_HTML_URL=[REDACTED]
CI_PULL_REQUEST_NUMBER=12345
CI_PULL_REQUEST_SOURCE_BRANCH=feature/jared/my-branch
CI_PULL_REQUEST_SOURCE_COMMIT=[REDACTED]
CI_PULL_REQUEST_SOURCE_REPO=[REDACTED]
CI_PULL_REQUEST_TARGET_BRANCH=feature/jared/my-branch-name
CI_PULL_REQUEST_TARGET_COMMIT=[REDACTED]
CI_PULL_REQUEST_TARGET_REPO=[REDACTED]
CI_RESULT_BUNDLE_PATH=/Volumes/workspace/resultbundle.xcresult
CI_START_CONDITION=pr_update
CI_TEAM_ID=[REDACTED]
CI_TEST_PLAN=AllTests
CI_TEST_PRODUCTS_PATH=/Volumes/workspace/TestProducts.xctestproducts
CI_WORKFLOW=PR Validation
CI_WORKSPACE_PATH=/Volumes/workspace
CI_XCODEBUILD_ACTION=build-for-testing
CI_XCODE_CLOUD=TRUE
CI_XCODE_PROJECT=MyApp.xcworkspace
CI_XCODE_SCHEME=MyApp
TMPDIR=/Volumes/workspace/tmp/
```

**Comment by jaredsinclair** [comment]
I filed the above [as a new issue](https://github.com/realm/SwiftLint/issues/5448) (#5448) because I believe the package plugin variant of using SwiftLint may have been overlooked during the earlier fixes for Xcode Cloud.

**Comment by SimplyDanny** [comment]
> I filed the above [as a new issue](https://github.com/realm/SwiftLint/issues/5448) (#5448) because I believe the package plugin variant of using SwiftLint may have been overlooked during the earlier fixes for Xcode Cloud.

#5287 has only been done for package plugins.
